### PR TITLE
Add 4ti2

### PR DESCRIPTION
**Pull request by saraedum**
an optional dependency of SageMath, see https://github.com/conda-forge/sage-feedstock/issues/42. For me it's a prerequisite to package sage-flatsurf.

### CODE REVIEW

**Comment by saraedum** [comment]
Would you mind becoming maintainers on this package as well @videlec @mkoeppe @isuruf? (This gives you access to the package in conda-forge so that you can fix things if something does not work right but does not come with any obligations.)

File: recipes/4ti2/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,40 @@
+{% set name = "4ti2" %}
+{% set version = "1.6.9" %}
+
+package:
+  name: {{ name }}
+  version: {{ version }}
+
+source:
+  url: https://github.com/4ti2/{{ name }}/archive/{{ version }}.tar.gz
```
Comment (isuruf) [inline recipes/4ti2/meta.yaml]:
```suggestion
  url: https://github.com/4ti2/4ti2/releases/download/Release_{{ version.replace(".", "_") }}/4ti2-{{ version }}.tar.gz
```

File: recipes/4ti2/build.sh [hunk 2]
```diff
@@ -0,0 +1,5 @@
+#!/bin/bash
+
+./configure
```
Comment (isuruf) [inline recipes/4ti2/build.sh]:
```suggestion
./configure --prefix=$PREFIX
```

File: recipes/4ti2/build.sh [hunk 3]
```diff
@@ -0,0 +1,5 @@
...
+make
```
Comment (isuruf) [inline recipes/4ti2/build.sh]:
```suggestion
make -j${CPU_COUNT}
```

**Comment by saraedum** [comment]
Thanks for fixing up my draft package :)

**Comment by saraedum** [comment]
I think this is ready to go. Just waiting for feedback from potential extra maintainers.

File: recipes/4ti2/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+    - automake
+    - libtool
+    - {{ compiler('c') }}
+    - {{ compiler('cxx') }}
+  host:
+    - gmp
+    - glpk
+  run:
+    # We link against glpk which does not specify a run_exports so we need to add it explicitly here.
+    - glpk
```
Comment (isuruf) [inline recipes/4ti2/meta.yaml]:
```suggestion
    - {{ pin_compatible("glpk", max_pin="x.x") }}
```

Can you send a PR for glpk to add run_exports?

Comment (isuruf) [reply recipes/4ti2/meta.yaml]: Also, can you send a PR to `conda-forge-pinning-feedstock`?

File: recipes/4ti2/build.sh [hunk 5]
```diff
@@ -0,0 +1,5 @@
+#!/bin/bash
+
+./configure --prefix=$PREFIX
+make -j${CPU_COUNT}
```
Comment (isuruf) [inline recipes/4ti2/build.sh]: Is there a `make check` ?

File: recipes/4ti2/meta.yaml (hunk 4 above)
Comment (saraedum) [reply recipes/4ti2/meta.yaml]: There's a pin in conda-forge-pinning-feedstock. So I don't think this is necessary.

File: recipes/4ti2/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+
+source:
+  url: https://github.com/4ti2/4ti2/releases/download/Release_{{ version.replace(".", "_") }}/4ti2-{{ version }}.tar.gz
+  sha256: 3053e7467b5585ad852f6a56e78e28352653943e7249ad5e5174d4744d174966
+
+build:
+  number: 0
+  skip: True  # [win]
+  run_exports:
+    - {{ pin_subpackage("4ti2") }}
```
Comment (isuruf) [inline recipes/4ti2/meta.yaml]: Why pin this? This package has only binaries right?

Comment (saraedum) [reply recipes/4ti2/meta.yaml]: You're right of course. I got confused by the `lib` directory.

**Pull request merged**